src/NuGet.Clients/NuGet.SolutionRestoreManager/RestoreOperationLogger.cs

```diff
@@ -145,10 +149,20 @@ public override void Log(ILogMessage logMessage)
                 // Avoid moving to the UI thread unless there is work to do
                 if (reportProgress || showAsOutputMessage)
                 {
+                    // Make sure the message is queued in order of calls to LogAsync, but don't wait for the UI thread
+                    // to actually show it.
+                    _loggedMessages.Enqueue(Tuple.Create(reportProgress, showAsOutputMessage, logMessage));
+
                     // Run on the UI thread
-                    Do((_, progress) =>
+                    var ignored = DoAsync((_, progress) =>
```

Comment:
➡️ This is a candidate feature for a future version of C# (dotnet/csharplang#111), but currently isn't available.